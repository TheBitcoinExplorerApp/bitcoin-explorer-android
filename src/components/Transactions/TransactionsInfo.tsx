import { View, Text, StyleSheet } from "react-native";
import useAppDataStore from "src/context/DataProvider";
import Transaction from "./component/Transaction";

const styles = StyleSheet.create({
  container: {
    gap: 15,
  },
  title: {
    fontSize: 15,
    fontWeight: "600",
    color: "white",
  },
});

export default function TransactionsInfo() {
  const { transactions, i18n } = useAppDataStore();

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{i18n.t("transactions")}</Text>

      <Transaction transactions={transactions} />
    </View>
  );
}
