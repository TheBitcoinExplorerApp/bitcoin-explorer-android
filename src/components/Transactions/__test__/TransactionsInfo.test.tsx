// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import TransactionsInfo from "../TransactionsInfo";

describe("TransactionsInfo", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TransactionsInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
