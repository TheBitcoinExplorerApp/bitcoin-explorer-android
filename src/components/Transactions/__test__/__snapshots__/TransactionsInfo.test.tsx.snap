// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsInfo renders correctly 1`] = `
<View
  style={
    {
      "gap": 15,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    Transações
  </Text>
  <View
    style={
      {
        "gap": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "backgroundColor": "#1d2133",
            "borderRadius": 7,
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "paddingVertical": 10,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 13,
                "fontWeight": "500",
              }
            }
          >
            ID Transação
          </Text>
          <Text
            style={
              {
                "color": "#DF7800",
                "fontWeight": "500",
              }
            }
          >
            0...
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 13,
                "fontWeight": "500",
              }
            }
          >
            Valor
          </Text>
          <Text
            style={
              {
                "color": "white",
                "fontWeight": "500",
              }
            }
          >
            0
             BTC
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 13,
                "fontWeight": "500",
              }
            }
          >
            Taxa
          </Text>
          <Text
            style={
              {
                "color": "white",
                "fontWeight": "500",
              }
            }
          >
            0
             sats
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "backgroundColor": "#1d2133",
            "borderRadius": 7,
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "paddingVertical": 10,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 13,
                "fontWeight": "500",
              }
            }
          >
            ID Transação
          </Text>
          <Text
            style={
              {
                "color": "#DF7800",
                "fontWeight": "500",
              }
            }
          >
            1...
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 13,
                "fontWeight": "500",
              }
            }
          >
            Valor
          </Text>
          <Text
            style={
              {
                "color": "white",
                "fontWeight": "500",
              }
            }
          >
            0
             BTC
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 13,
                "fontWeight": "500",
              }
            }
          >
            Taxa
          </Text>
          <Text
            style={
              {
                "color": "white",
                "fontWeight": "500",
              }
            }
          >
            0
             sats
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
